openstf/model/serializer.py

```diff
@@ -0,0 +1,204 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+from abc import ABC, abstractmethod
+from datetime import datetime
+from pathlib import Path
+
+import joblib
+import structlog
+from sklearn.base import RegressorMixin
+
+from ktpbase.config.config import ConfigManager
+
+
+MODEL_FILENAME = "model.sav"
+FOLDER_DATETIME_FORMAT = "%Y%m%d%H%M%S"
+
+
+class AbstractSerializer(ABC):
+    def __init__(self, prediction_job: dict) -> None:
+        self.pj = prediction_job
+        self.logger = structlog.get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    def save_model(self, model: RegressorMixin) -> None:
+        """Persists trained sklearn compantible model
+
+        Args:
+            model: Trained sklearn compatible model object
+        """
+        self.logger.error("This is an abstract method!")
+        pass
+
+    @abstractmethod
+    def load_model(self) -> RegressorMixin:
+        """Loads model that has been trained earlier
+
+        Returns: Trained sklearn compatible model object
+
+        """
+        self.logger.error("This is an abstract method!")
+
+
+class PersistentStorageSerializer(AbstractSerializer):
+
+    trained_models_folder = Path(ConfigManager.get_instance().paths.trained_models)
+
+    def save_model(self, model: RegressorMixin) -> None:
+        """Serializes trained sklearn compantible model to persistent storage
+
+        Args:
+            model: Trained sklearn compatible model object
+        """
+
+        # Compose save path
+        save_path = self._build_save_folder_path()
+
+        # Create save path if nescesarry
+        save_path.mkdir(parents=True, exist_ok=True)
+
+        # Save model
+        try:
+            joblib.dump(model, save_path / MODEL_FILENAME)
+        except Exception as e:
```

Comment:
Should this exception caught here on at the calling code? What type of things can typically go wrong?

Reply:
yes you are right, probably nothing goes wrong and if so then it is better to catch the exception higher up.